hledger-lib/Hledger/Read/Common.hs

```diff
@@ -194,6 +195,14 @@ rawOptsToInputOpts rawopts = InputOpts{
 
 --- * parsing utilities
 
+-- | Parse special separator names TAB and SPACE, or return the first
+-- character. Return Nothing on empty string
+getSpecialSeparators :: String -> Maybe Char
+getSpecialSeparators "SPACE" = Just ' '
+getSpecialSeparators "TAB" = Just '\t'
+getSpecialSeparators (x:_) = Just x
```

Comment:
We could do error handling here and split off this case from the `x:[]`. But the return signature would have to be IO (Maybe Char) because we're throwing exceptions.